App/App.jsx

```diff
@@ -0,0 +1,299 @@
+            (article) => article._id !== newsData._id
+          );
+          setSavedArticles(unsavedNewsArticles);
+        })
+        .catch((err) => {
+          console.log(err);
+        });
+    };
+
+    const handleSaveArticle = ({ newsData, keyword }) => {
+      if (!savedArticles.find((article) => article.link === newsData.url)) {
+        addSavedArticle(newsData, keyword)
+          .then((res) => {
+            setSavedArticles([res, ...savedArticles]);
+            const savedArticlesId = res._id;
+            const newArticle = { ...newsData, _id: savedArticlesId };
+            const newSearchResult = searchResult.map((article) =>
+              article.url === newsData.url ? newArticle : article
+            );
+            setSearchResult(newSearchResult);
+          })
+          .catch((err) => console.error(err));
+      } else if (savedArticles.some((article) => article.link === newsData.url)) {
+        removeSavedArticle(newsData)
+          .then(() => {
+            const unsaveNewsArticles = savedArticles.filter(
+              (article) => article._id !== newsData._id
+            );
+            setSavedArticles(unsaveNewsArticles);
+
+            const newArticle = { ...newsData, _id: "" };
+            const newSearchResult = searchResult.map((article) =>
+              article.url === newsData.url ? newArticle : article
+            );
+            setSearchResult(newSearchResult);
+          })
+          .catch((err) => console.error(err));
+      }
+    };
+
+    useEffect(() => {
+      const jwt = tokenValue.getToken(); //getting token from localStorage
+      if (!jwt) {
+    return;
+  }
+      return getUserInfo(jwt)
+        .then((res) => {
+          if (res) {
+              setCurrentUser({
+             
+            name: res.data.username,
+            email: res.data.email,
+            _id: res.data._id,
+              });
+             setIsLoggedIn(true);
+            getSavedArticles(jwt)
+              .then((res) => {
+              
+                setSavedArticles(res);
+              })
+              .catch((err) => {
+                console.log(err);
+              });
+          }
+        })
+        .catch((err) => {
+          console.log(err);
+        })
+        .finally(() => { });
+    }, []);
+
+    useEffect(() => {
+      const handleEscClose = (e) => {
+        if (e.key === "Escape") {
+          onClose();
+        }
+      };
+      document.addEventListener("keydown", handleEscClose);
+      return () => {
+        document.removeEventListener("keydown", handleEscClose);
+      };
+    }, []);
+
+    useEffect(() => {
+      const handleOverlayClick = (e) => {
+        if (e.target.classList.contains("modal")) {
+          onClose();
+        }
+      };
+      document.addEventListener("click", handleOverlayClick);
+      return () => {
+        document.removeEventListener("click", handleOverlayClick);
+      };
+    }, []);
```

Comment:
You should add an `overlay` listener like this to be able to remove it **correctly** when modals are closed:

```js
  }, [activeModal]);  // watch activeModal here
```